Cargo.toml

```diff
@@ -36,6 +36,95 @@ installers = []
 # Target platforms to build apps for (Rust target-triple syntax)
 targets = ["x86_64-unknown-linux-musl", "aarch64-unknown-linux-musl"]
 
+[workspace.dependencies]
+bottlerocket-types = { version = "0.0.14", git = "https://github.com/bottlerocket-os/bottlerocket-test-system", tag = "v0.0.14" }
+bottlerocket-variant = { version = "0.1", path = "tools/bottlerocket-variant" }
+buildsys = { version = "0.1", path = "tools/buildsys", lib = true, artifact = [ "bin:buildsys", "bin:bottlerocket-variant" ] }
+buildsys-config = { version = "0.1", path = "tools/buildsys-config" }
+oci-cli-wrapper = { version = "0.1", path = "tools/oci-cli-wrapper" }
+parse-datetime = { version = "0.1", path = "tools/parse-datetime" }
+pipesys = { version = "0.1", path = "tools/pipesys", lib = true, artifact = [ "bin:pipesys" ] }
```

Comment:
What does `lib = true` do here? It's sort of surprising since these are all binaries.

Can we move the artifact dependency stuff to `twoliter/Cargo.toml` and leave just the version and path here? It seems a little weird for cases where we have a cross crate dependency (like `pubsys` depends on `buildsys`).

Reply:
> What does lib = true do here? It's sort of surprising since these are all binaries.

`buildsys` depends on `pipesys` as a library, and you need to add `lib = true` if you use artifact dependencies to tell Cargo to build both.
```
   Compiling buildsys v0.1.0 (/home/ec2-user/twoliter/tools/buildsys)
error[E0433]: failed to resolve: use of undeclared crate or module `pipesys`
  --> tools/buildsys/src/builder.rs:20:5
   |
20 | use pipesys::server::Server as PipesysServer;
   |     ^^^^^^^ use of undeclared crate or module `pipesys`
```

> Can we move the artifact dependency stuff to `twoliter/Cargo.toml` and leave just the version and path here?

I really wanted to do it this way, but Cargo wouldn't cooperate. e.g. if you move the `artifact` declaration for `pubsys` to `twoliter/Cargo.toml`, `cargo build` says:

```
cargo clippy --locked -- -D warnings --no-deps
warning: /home/ec2-user/twoliter/twoliter/Cargo.toml: unused manifest key: dependencies.pubsys.artifact
warning: twoliter v0.4.4 (/home/ec2-user/twoliter/twoliter) ignoring invalid dependency `pubsys` which is missing a lib target
```

... and then eventually

```
    Checking pubsys v0.1.0 (/home/ec2-user/twoliter/tools/pubsys)
error: environment variable `CARGO_BIN_FILE_PUBSYS` not defined at compile time
  --> twoliter/src/tools.rs:17:38
   |
17 | const PUBSYS: &[u8] = include_bytes!(env!("CARGO_BIN_FILE_PUBSYS"));
   |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Cargo sets build script variables at run time. Use `std::env::var("CARGO_BIN_FILE_PUBSYS")` instead
   = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `twoliter` (bin "twoliter") due to 1 previous error
make: *** [Makefile:14: clippy] Error 101
```